query-service-impl/src/test/java/org/hypertrace/core/query/service/pinot/ExecutionContextTest.java

```diff
@@ -308,4 +311,133 @@ public void testSelectionsLinkedHashSet() {
     assertEquals(
         Expression.newBuilder().setFunction(minFunction).build(), selectionsIterator.next());
   }
+
+  @Test
+  public void testSetTimeSeriesPeriod() {
+
+    // no group by
+    Builder builder = QueryRequest.newBuilder();
+    QueryRequest queryRequest = builder.build();
+    ExecutionContext context = new ExecutionContext("test", queryRequest);
+    assertEquals(Optional.empty(), context.getTimeSeriesPeriod());
+
+    // group by on column
+    builder.addGroupBy(
+        Expression.newBuilder()
+            .setColumnIdentifier(
+                ColumnIdentifier.newBuilder().setColumnName("Trace.service_name")));
+    queryRequest = builder.build();
+    context = new ExecutionContext("test", queryRequest);
+    assertEquals(Optional.empty(), context.getTimeSeriesPeriod());
+
+    // group by on other functions
+    Function count =
+        Function.newBuilder()
+            .setFunctionName("Count")
+            .setAlias("myCountAlias")
+            .addArguments(
+                Expression.newBuilder()
+                    .setColumnIdentifier(ColumnIdentifier.newBuilder().setColumnName("Trace.id")))
+            .build();
+    builder.addGroupBy(Expression.newBuilder().setFunction(count));
+    queryRequest = builder.build();
+    context = new ExecutionContext("test", queryRequest);
+    assertEquals(Optional.empty(), context.getTimeSeriesPeriod());
+  }
+
+  @Test
+  public void testSetTimeSeriesPeriodForTimeSeriesRequest() {
+    Builder builder = QueryRequest.newBuilder();
+    builder.addGroupBy(
+        Expression.newBuilder()
+            .setFunction(createTimeColumnGroupByFunction("SERVICE.startTime", 15))
+            .build());
+    QueryRequest queryRequest = builder.build();
+    ExecutionContext context = new ExecutionContext("test", queryRequest);
+    assertEquals(Optional.of(Duration.ofSeconds(15)), context.getTimeSeriesPeriod());
+  }
+
+  @Test
+  public void testComputeTimeRangeDuration() {
+    QueryRequest queryRequest = getQueryRequestWithTimeFilter();
+    ExecutionContext context = new ExecutionContext("test", queryRequest);
+    context.setTimeFilterColumn("SERVICE.startTime");
+    context.computeTimeRangeDuration(queryRequest);
+    assertEquals(Duration.ofSeconds(3600), context.getTimeRangeDuration());
```

Comment:
This test isn't readable without reading what `getQueryRequestWithTimeFilter` returns. Since we're testing that the time filter calculation is correct, we should show in this test what we set the time filter to (i.e. have the query request building function take in time range start and end)